Scalar.Common/Enlistment.cs

```diff
@@ -36,15 +36,22 @@ public abstract class Enlistment
                 GitProcess.ConfigResult originResult = gitProcess.GetOriginUrl();
                 if (!originResult.TryParseAsString(out string originUrl, out string error))
                 {
-                    throw new InvalidRepoException(this.WorkingDirectoryRoot, "Could not get origin url. git error: " + error);
+                    if (!gitProcess.TryGetRemotes(out string[] remotes, out error))
+                    {
+                        throw new InvalidRepoException(this.WorkingDirectoryRoot, $"Failed to load remotes with error: {error}");
+                    }
+
+                    if (remotes.Length > 0)
```

Comment:
Should this be `remotes.Length == 1`?  If there are more that one are we sure we are doing the right thing to just take the first remote?  Should there be a config setting for that case?

Reply:
I thought it safer to have _a_ RepoUrl setting rather than have none.

I think it is important that we will not support multiple remotes when using the GVFS protocol, and I think that's the only time we use RepoUrl (other than logging).